src/actions/complaints.jsx

```diff
@@ -1,18 +1,106 @@
-import * as types from '../constants'
+import { MODE_LIST, MODE_MAP, MODE_TRENDS } from '../constants'
 
+export const AGGREGATIONS_API_CALLED = 'AGGREGATIONS_API_CALLED'
+export const AGGREGATIONS_RECEIVED = 'AGGREGATIONS_RECEIVED'
+export const AGGREGATIONS_FAILED = 'AGGREGATIONS_FAILED'
+export const COMPLAINTS_API_CALLED = 'COMPLAINTS_API_CALLED'
 export const COMPLAINTS_RECEIVED = 'COMPLAINTS_RECEIVED'
 export const COMPLAINTS_FAILED = 'COMPLAINTS_FAILED'
 export const COMPLAINT_DETAIL_RECEIVED = 'COMPLAINT_DETAIL_RECEIVED'
 export const COMPLAINT_DETAIL_FAILED = 'COMPLAINT_DETAIL_FAILED'
+export const STATES_API_CALLED = 'STATES_API_CALLED'
+export const STATES_RECEIVED = 'STATES_RECEIVED'
+export const STATES_FAILED = 'STATES_FAILED'
+
+// ----------------------------------------------------------------------------
+// Routing action
+/**
+ * Routes to the correct endpoint based on the state
+ *
+ * @returns {Promise} a chain of promises that will update the Redux store
+ */
+export function sendQuery() {
+  // eslint-disable-next-line complexity
+  return ( dispatch, getState ) => {
+
+    const state = getState()
+    const viewMode = state.query.tab
+    switch ( viewMode ) {
+      case MODE_MAP:
+      case MODE_LIST:
+      case MODE_TRENDS:
+        // eslint-disable-next-line no-warning-comments
+        // TODO: this should be getAggregations once the API is built
+        // dispatch( getAggregations() )
+        dispatch( getComplaints() )
+        break
+      default:
+        return
+    }
+
+    // Send the right-hand queries
+    dispatch( sendHitsQuery() )
+  }
+}
+
+/**
+ * Routes to the correct endpoint based on the state
+ *
+ * @returns {Promise} a chain of promises that will update the Redux store
+ */
+export function sendHitsQuery() {
+  // eslint-disable-next-line complexity
+  return ( dispatch, getState ) => {
+    const state = getState()
+    const viewMode = state.query.tab
+    switch ( viewMode ) {
+      case MODE_MAP:
+        dispatch( getStates() )
+        break
+      // case 'Trends':
+      //   dispatch( getTrends() )
+      //   break
+      case MODE_LIST:
+        dispatch( getComplaints() )
```

Comment:
The way this reads, `getComplaints()` is called twice for `MODE_LIST` when `sendQuery()` is called

Reply:
Yes, sorry, that's intentional.  I put getComplaints in there as a placeholder for the aggregations for now and didn't want to have to change it again when the api endpoint is available.